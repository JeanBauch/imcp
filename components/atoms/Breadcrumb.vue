<template>
	<nav id="breadcrumb">
		<ol class="flex">
			<li
				v-for="(item, index) in crumbs"
				:key="index"
			>
				<nuxt-link :to="`/${item.url}`">
					<p class="text-sm font-light inline-block pr-7 relative after:content-['>']" :class="(crumbs.length - 1 == index )? 'text-gray after:content-[i]' : 'after:align-bottom after:absolute after:bottom-0 after:right-3'">
						{{ item.name }}
					</p>
				</nuxt-link>
			</li>
		</ol>
	</nav>
</template>

<script lang="ts">
import Vue from 'vue';

export default Vue.extend({
	props: {
		crumbs: {
			type: Array,
			required: true,
			default: () => [],
		},
	},
});
</script>
