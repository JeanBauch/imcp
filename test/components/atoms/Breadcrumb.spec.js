import { shallowMount, RouterLinkStub } from '@vue/test-utils';

import Breadcrumb from '~/components/atoms/Breadcrumb.vue';

describe('Components:atoms:Breadcrumb', () => {
	describe('Snapshots', () => {
		it('should render component with props data', () => {
			const wrapper = shallowMount(Breadcrumb, {
				propsData: {
					crumbs: [{
						name: 'Inicio',
						url: '',
					}, {
						name: 'Noticias',
						url: 'noticias',
					}],
				},
				stubs: {
					NuxtLink: RouterLinkStub,
				},
			});

			expect(wrapper.html()).toMatchSnapshot();
		});

		it('should render component without props data', () => {
			const wrapper = shallowMount(Breadcrumb, {
				propsData: {
					crumbs: [],
				},
				stubs: {
					NuxtLink: RouterLinkStub,
				},
			});

			expect(wrapper.html()).toMatchSnapshot();
		});

		it('should coverage default props factory function', () => {
			const wrapper = shallowMount(Breadcrumb, {
				propsData: {
					crumbs: [],
				},
				stubs: {
					NuxtLink: RouterLinkStub,
				},
			});

			expect(wrapper.vm.$options.props.crumbs.default.call()).toEqual([]);
		});
	});
});
