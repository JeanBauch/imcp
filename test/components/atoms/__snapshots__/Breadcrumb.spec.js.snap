// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components:atoms:Breadcrumb Snapshots should render component with props data 1`] = `
"<nav id=\\"breadcrumb\\">
  <ol class=\\"flex\\">
    <li><a>
        <p class=\\"text-sm font-light inline-block pr-7 relative after:content-['>'] after:align-bottom after:absolute after:bottom-0 after:right-3\\">
          Inicio
        </p>
      </a></li>
    <li><a>
        <p class=\\"text-sm font-light inline-block pr-7 relative after:content-['>'] text-gray after:content-[i]\\">
          Noticias
        </p>
      </a></li>
  </ol>
</nav>"
`;

exports[`Components:atoms:Breadcrumb Snapshots should render component without props data 1`] = `
"<nav id=\\"breadcrumb\\">
  <ol class=\\"flex\\"></ol>
</nav>"
`;
